### PR TITLE
Allow custom amplicon master file upload

### PR DESCRIPTION
**Pull request by jkokosar**

### CODE REVIEW

File: resolwe_bio/tests/processes/test_amplicon.py [hunk 1]
```diff
@@ -13,7 +13,7 @@ def test_bwa_trim(self):
         reads = self.run_process('upload-fastq-paired', inputs)
 
         genome = self.run_process('upload-genome', {'src': 'hs_b37_chr2_small.fasta.gz'})
-        master_file = self.run_process('upload-master-file', {'src': '56G_masterfile_test.txt'})
+        master_file = self.prepare_amplicon_master_file
```
Comment (mzganec) [inline resolwe_bio/tests/processes/test_amplicon.py]: This is a function, add parentheses at the end.

File: resolwe_bio/tools/validate_master_file.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    parser = argparse.ArgumentParser(description="Run gene ontology enrichment analysis.")
+    parser.add_argument('master_file', type=str, help="Master file.")
+    return parser.parse_args()
+
+
+def check_dna_sequence(sequence):
+    """Check if a given sequence contains only the allowed letters A, C, T, G."""
+    for base in sequence:
+        if base in ['A', 'C', 'T', 'G']:
+            return True
```
Comment (mzganec) [inline resolwe_bio/tools/validate_master_file.py]: This is likely not what you intended to do, the `for` loop always stops at the first base.

File: resolwe_bio/processes/import_data/master_file.yml [hunk 3]
```diff
@@ -4,13 +4,16 @@
 ---
 
 - slug: upload-master-file
-  name: Upload master file
+  name: Upload custom file
```
Comment (mzganec) [inline resolwe_bio/processes/import_data/master_file.yml]: Could be `Upload custom master file`.

**Comment by jkokosar** [comment]
@mzganec Please make another review pass at your convenience. Some additional changes were pushed after your latest review.

**Pull request merged**